zkml/docs/src/end_to_end_lu.md

```diff
@@ -0,0 +1,28 @@
+# End-to-End Lookup Protocol
+Since a model will have many repeated layers to improve efficiency we aim to only produce one multiplicity polynomial per table type. An added bonus to this is that at each step we only have to do proving work proportional to the number of lookups (for lookup steps) or the table size (for mulitpilcity polynomials).
+
+## Splitting Lookups and Tables
+As a quick recap the LogUp protocol works as follows. If we have a table $` T:=\{t_{i} \}_{i = 1}^{m} `$ and a set of values $` L:= \{lk_{i}\}_{i=1}^{n} `$ to be looked up then the LogUp protocol proves 
+
+$$\begin{align} \sum_{i = i}^{m}\frac{M_{i}}{\alpha + t_{i}} = \sum_{j=1}^{n}\frac{1}{\alpha + lk_{j}}.  \end{align}$$
+
+Where $`\alpha`$ is a random challenge provided by the verifier and the set $ M $ is defined as
+
+$$\begin{align} M := \{\mathrm{count}(lk_{j})/\mathrm{count}(t_{i}) | t_{i} = lk_{j}\}, \end{align}$$
+
+the number of times the value appears in $` L `$ divided by the number of times it appears in $` T `$. The indexing of $` M `$ lines up with the indexing of $T $, by this we mean $` M_{i} \in M `$ is the number of times $`t_{i} \in T `$ is looked up.
+
+ We use a GKR circuit to prove correct computation of both sides of Equation $`(1)`$  separately. The output of this computation for the right hand side of Equation $`(1)`$ is the claimed numerator and denominator, $` p `$ and $` q `$, such that 
+
+$$ \begin{align}\sum_{j=1}^{n}\frac{-1}{\alpha + lk_{j}} = \frac{p}{q}. \end{align}$$
+
+Like wise the circuits for the left hand side output $` r`$ and $`s`$ such that
+
+$$\begin{align} \sum_{i=1}^{m}\frac{M_{i}}{\alpha + t_{i}} = \frac{r}{s}. \end{align}$$
+
+At the end of the entire protocol we have GKR proofs, together with claimed numerators and denominators for each of the lookups and the tables. The verifier runs the GKR verification procedure for each of the lookup/table proofs and, provided they are all correct, checks that their combined numerator is zero and that their combined denominator is non-zero.
```

Comment:
Ok you explain logup gkr but can you explain the trick you're doing with the multiplicity polynomial ?